### ISSUE TITLE
[Security] Check for Validity/Legitimation of EU DCCs (and "force-update" apps)

### ISSUE THREAD

**Issue by vaubaehn** [Digitaler-Impfnachweis/covpass-android#96]
## Avoid duplicates
* [X] Question is not already answered in the [FAQ](https://digitaler-impfnachweis-app.de/faq/)
* [X] Question has not already been asked in another issue

*THIS QUESTION/ISSUE ALSO AFFECTS iOS VERSION OF COVPASS*

## Your Question
Hi there,
I suppose you are already aware of some bad things going on that are discussed here: https://github.com/ehn-dcc-development/hcert-spec/discussions/105
To stop/decrease the spread of fraudulent DCCs, it is necessary to invalidate them - at least with the official checking apps, and CovPassCheck is already doing its job here 👍 . But an additional factor to make the use of fraudulent DCCs more difficult, is to invalidate them also in the wallet apps. While Corona-Warn-App versions >=2.7 can do this, versions <2.7 have a problem.
(see also https://github.com/corona-warn-app/cwa-documentation/issues/732 and https://github.com/ehn-dcc-development/hcert-spec/discussions/105#discussioncomment-1566196)

So, my questions go here:
- Does CovPass check validity/signature/legitimation of DCCs since its very first release? Or has that functionality been introduced later? Discussing this with @Ein-Tim , he remembers it to be integrated since the very first CovPass version. Probably he is true, but I still have a feeling that it was introduced later...
- In case the signature/legitimation check was introduced later, did you implement a kill switch or a forced update mechanism that older CovPass versions may practically be disabled and "common users" will not have the ability to present unlegitimate DCCs to gate keepers? (Yes, gate keepers should use CovPassCheck, but currently this is rare practice, unfortunately...)

Thanks in advance for your responses!

**Comment by alexcimander** [comment]
Hey  vaubaehn,

the signature/ legitimation check was there from beginning on but the need for invalidation is definitely growing since the number of certificates issued with criminal intent is increasing. For that we are working on EU wide long term approaches as well as short term approaches for our apps. Those mechanisms should prevent fraudulent certificates to be usable (in the CovPassCheck as well as the CovPass).

For now no force-update is in place but we are discussing ways to make sure that the mechanisms take effect, force-update might be one of them.

We will keep you updated.

Thank you for your questions and the open discussion :-)

**Comment by eku** [comment]
Hallo @alexcimander 
How do you think the `force-update` can work on devices without Google services? I mean specifically the version of the apps from F-Droid.

**Comment by vaubaehn** [comment]
Hi @alexcimander ,
thanks for your kind reply! Good to hear!

> We will keep you updated.

Will this be though this issue? Or should this one be closed here?

**Comment by vaubaehn** [comment]
> How do you think the `force-update` can work on devices without Google services? I mean specifically the version of the apps from F-Droid.

@eku 
I'm not one of the devs from here, but Google services are practically not necessary. The app itself can be informed via remote app config and just display a nag screen (or something more elegant) to urge users to update.

**Comment by treysis** [comment]
I am still getting valid certificate checks from CovPassCheck for the infamous fraudulent certificates. My database is updated (shows date of Nov 13, 2021).

**Comment by alexcimander** [comment]
Hey everyone, thank you for the active discussion!

First of "force-update" found its way into the backlog and we will inform you about updates of the possible implementation 🚧 

By now all affected certificates which we were advised to invalidate should be blocked by now (Release 1.13).
These certificates won't be accepted by the check-app anymore and have a restricted usability in the CovPass.

Thank you everyone! 👍

**Comment by jleufgen** [comment]
@vaubaehn could we close this ticket?

**Comment by vaubaehn** [comment]
Hi @jleufgen ,

as Alex Cimander wrote

> First of "force-update" found its way into the backlog and we will inform you about updates of the possible implementation 🚧

I would suggest to leave the issue open, until a 'force-update' mechanism is implemented. I understood his statement, that any update from your side will be commented here.

So, would it be fine with you to leave it open for now?

Kind regards

**Comment by pdolinic** [comment]
Well if there is no revocation-list and there are private-keys floating around this will end badly?

**Comment by treysis** [comment]
Nope, no keys floating around. Just nobody seems to remove the compromised `kid`s from the list? Italy and Switzerland did. So no idea what's going on in Germany.

**Comment by ingowerren** [comment]
In 2021, the European Union introduced the digital COVID certificate as an EU-wide recognised proof of vaccination, recovery or negative testing for COVID-19. This regulation expired on 30 June 2023. On a transitional basis, the EU will continue to operate the technical systems to enable cross-border certificate checks until 31 December 2023. The transitional operation will end on that date.
In Germany, the possibility of issuing digital COVID certificates such as vaccination and recovery certificates will therefore be discontinued on 31 December 2023. The CovPass app and the CovPassCheck app will therefore also be discontinued. The CovPass app will be switched to wallet mode, where the stored certificates will be retained.

What does this mean for you?
All functionalities will continue to beavailable until 31 December 2023.

**From 1 January 2024:**

Support for the app will no longer be offered.

- The apps will no longer be available for (re)download in the app store.
- No new functions will be published.
- Further information can also be found in the CovPass app privacy policy.
